lib/cli.js

```diff
@@ -278,26 +279,47 @@ exports.process = () => {
 }
 
 exports.run = () => {
-  const config = exports.process()
-
-  switch (config.cmd) {
-    case 'start':
-      new Server(config).start()
-      break
-    case 'run':
-      require('./runner')
-        .run(config)
-        .on('progress', printRunnerProgress)
-      break
-    case 'stop':
-      require('./stopper').stop(config)
-      break
-    case 'init':
-      require('./init').init(config)
-      break
-    case 'completion':
-      require('./completion').completion(config)
-      break
+  const cliOptions = exports.process()
+  const cmd = cliOptions.cmd // prevent config from changing the command
+  const cmdNeedsConfig = cmd === 'start' || cmd === 'run' || cmd === 'stop'
+  if (cmdNeedsConfig) {
+    cfg.parseConfig(
+      cliOptions.configFile,
+      cliOptions,
+      {
+        promiseConfig: true,
+        throwErrors: true
+      }
+    ).then(function onKarmaConfigFulfilled (config) {
+      switch (cmd) {
+        case 'start':
+          new Server(config).start()
+          break
+        case 'run':
+          require('./runner')
+            .run(config)
+            .on('progress', printRunnerProgress)
+          break
+        case 'stop':
+          require('./stopper').stop(config)
+          break
+      }
+    }, function onKarmaConfigRejected (reason) {
+      console.error(reason) // TODO: configure a CLI Logger?
```

Comment:
Should this be removed? The way `fail` turned out, a logger is already called with a message that is almost the same a `reason`.

https://github.com/karma-runner/karma/blob/c4fd16e88ccdc79410ed579a32add267b6789e30/lib/config.js#L364-L371

Reply:
Yes, please!